docs/discounts-lists/index.md

```diff
@@ -0,0 +1,496 @@
+However, we get a compilation error:
+
+```text
+File "docs/order-confirmation/Discount.re", line 61, characters 18-19:
+61 |   |> List.nth_opt(1);
+                       ^
+Error: This expression has type int but an expression was expected of type
+         'a list
+```
+
+## Placeholder operator
+
+This is because of the type signature of `List.nth_opt` is
+
+```text
+list('a) => int => option('a)
+```
+
+That is, it accepts the list as the first argument, not the last. Recall that
+the pipe last operator (`|>`) pipes values into the last argument of a function.
+However, there's a way to override the placement of the argument:
+
+```reason
+|> List.nth_opt(_, 1)
+```
+
+Do not confuse the `_` here for wildcard, here it's a placeholder for where the
+argument should go. When we put `_` in the first argument position, it overrides
+the default behavior of the pipe last operator. Run `npm run test` to confirm
+that `Discount.getFreeBurger` works the same as before.
+
+# `List.nth_opt` is unsafe
+
+The problem with using `List.nth_opt` is that it can still raise an
+`Invalid_argument` exception if the value of `n` is negative. You can confirm
+this by hovering over `nth_opt` and reading the popup or [read its
+documentation](https://melange.re/v1.0.0/api/re/melange/Stdlib/List/#val-nth_opt).
+While this is unlikely to cause a problem inside `Discount.getFreeBurger`, it's
+best to avoid unsafe functions except in special circumstances, for example if a
+function needs to be as fast as possible. Of course, it's possible to implement
+a completely safe version of `nth_opt`, which is an exercise at the end of this
+chapter.
+
+---
+
+Mazel tov! You've implemented the burger discounts in a way that is more
+maintainable, and you've also learned a lot about lists along the way. In the
+next chapter, we'll finally use the discount logic to reduce the final price of
+an order.
+
+## Overview
+
+- Lists are immutable
+- You can pattern match on a whole list, even if you don't know its length
+- The `List` module contains most of the functions you'll need for
+  dealing with lists
+- The runtime representation of lists:
+  - Empty list → `0`
+  - Nonempty list → a JavaScript object with the fields `hd` (for head) and `tl`
+    (for tail)
+- The delimiters for list literals are `[]`
+- You can use list spread syntax (`...`) to add elements to the front of a list
+  or pattern match on the tail of a list
+  - The names of equivalent functions might not match the names in `Js.Array`
+- Documentation comments show up in editor hover popups and generated
+  documentation pages
+  - They can be attached to functions, modules, types, and
+  variables
+- The `ListLabels` module contains the same functions as in `List`, but they
+  have labeled arguments instead of positional arguments
+- The placeholder operator (`_`) can be used to override the position of the
+  piped argument when using the pipe last operator
```

Comment:
If you decide to follow the comment above, this section would have to be removed.